desc/protoparse/parser.go

```diff
@@ -412,6 +420,10 @@ func parseProtoFile(acc FileAccessor, filename string, importLoc *SourcePos, err
 			}()
 			result = parseProto(filename, in, errs, validate)
 		}()
+	} else if d, ok := p.FileDescriptors[filename]; ok {
```

Comment:
Hmmm. Shouldn't we consult this map _before_ trying to invoke `acc`? Or is it expected that source files provided by `acc` would override any explicit dependencies provided? I'm unclear exactly how this should work.

Note, we should also tighten up the spec in the doc comment for this field. My personal opinion is that the give descriptors are used if they exist, then we look for source code, and then finally we fall back to well-known/standard imports.

Reply:
My intuition is that the priority order should be:

1. `acc`
2. Imported Descriptors
3. Well-known protos (`google.protobuf.*`)

It seems like if the accessor finds a file (usually on disk), then that file probably gets preference over the Go package. But I have no justification for this claim other than raw intuition. I am happy to reverse the order of (1) and (2) if you are unconvinced.

Reply:
Thinking about it more, I think what you suggest (and what is implemented) is fine. But please document it in the Go docs.

Reply:
Ack, will do.

Reply:
Done.